internal/pkg/api/error.go

```diff
@@ -167,7 +167,7 @@ func NewHTTPErrResp(err error) HTTPErrResp {
 		{
 			ErrAgentIdentity,
 			HTTPErrResp{
-				http.StatusBadRequest,
+				http.StatusForbidden,
```

Comment:
I'm pretty sure I changed all the relevant auth errors to return a 401 here.

However I think it makes sense to use a 403 instead of a 401 in this case where the API key does not match the requested agent. I've altered the openapi spec to include it as a return for the relevant endpoints

cc @joshdover

Reply:
Makes sense to me.
So IIUC if the key doesn't match it will be a `403` everything else will be `401`.

If the key is not found, shouldn't we also return a `403` instead of a `401`?

Reply:
> If the key is not found, shouldn't we also return a `403` instead of a `401`?

This would be an unauthorized case, right? So 401 would be fine.